bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/deferred/LazySortedCollection.java

```diff
@@ -386,7 +387,9 @@ private final int partition(int subTree, FastProgressReporter mon) throws Interr
 			}
 
 			if (mon.isCanceled()) {
-				throw new InterruptedException();
+				InterruptedException interruptedException = new InterruptedException();
+				interruptedException.initCause(new OperationCanceledException());
+				throw interruptedException;
```

Comment:
Like above: semantically incorrect and unnecessary. The exception is only used for internal control flows of the class and not passed outside through public API, so no need to introduce this overhead.